testutil/testutil-core/src/main/java/com/sap/cloud/sdk/testutil/PrincipalContextApi.java

```diff
@@ -0,0 +1,29 @@
+package com.sap.cloud.sdk.testutil;
+
+import com.sap.cloud.sdk.cloudplatform.security.principal.DefaultPrincipal;
+import com.sap.cloud.sdk.cloudplatform.security.principal.Principal;
+import com.sap.cloud.sdk.cloudplatform.security.principal.PrincipalThreadContextListener;
+
+public interface PrincipalContextApi extends TestContextApi
+{
+    default Principal setPrincipal()
+    {
+        return setPrincipal("principal");
```

Comment:
[minor]

Same remark as for the `TenantContentApi`.

```suggestion
        return setPrincipal("default-test-principal");
```